File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res={{1}};
        for(int i=1;i<numRows;i++){
            vector<int> ans={1};
            for(int j=1;j<=i-1;j++){
                ans.push_back(res[i-1][j-1]+res[i-1][j]);
            }
            ans.push_back(1);
            res.push_back(ans);
        }
        return res;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        # n = len(numRows)
        if numRows==1: return [[1]]
        if numRows==2: return [[1],[1,1]]
        res = [[1],[1,1]]
        for row in range(2,numRows):
            temp = [1]
            # update list 
            for col in range(1,row):
                val = res[row-1][col-1]+res[row-1][col]
                temp.append(val)
            temp.append(1)
            res.append(temp)
        return res